### PR TITLE
#697 Enhance error message text in GTP.

### PR DESCRIPTION
**Pull request by barrybecker4**
Give the user more feedback if they incorrectly type a GTP command.

### CODE REVIEW

File: src/GTP.cpp [hunk 1]
```diff
@@ -265,15 +265,16 @@ bool GTP::execute(GameState & game, std::string xinput) {
 
         if (!cmdstream.fail()) {
             if (tmp != 19) {
-                gtp_fail_printf(id, "unacceptable size");
+                gtp_fail_printf(id,
```
Comment (gcp) [inline src/GTP.cpp]: This violates the GTP protocol, see: 
https://www.lysator.liu.se/~gunnar/gtp/gtp2-spec-draft2/gtp2-spec.html#sec:list-of-all-commands

fails | Syntax error. If the engine cannot handle the new                 size, fails with the error message "unacceptable size".

It's the job of the GTP interface to translate these error messages to something more user friendly.

Using newlines in the responses seems to me like you risk breaking some GTP interfaces that don't expect this. I would avoid it if possible.

File: src/GTP.cpp [hunk 2]
```diff
@@ -704,18 +717,17 @@ bool GTP::execute(GameState & game, std::string xinput) {
         } else if (winner_color == "b" || winner_color == "black") {
             who_won = FullBoard::BLACK;
         } else {
-            gtp_fail_printf(id, "syntax not understood");
+            gtp_fail_printf(id, 
+                "Unexpected winner color: %s.\n"
```
Comment (gcp) [inline src/GTP.cpp]: These commands are only produced by automated tools like GTP, so there's no point to this.

File: src/GTP.cpp [hunk 3]
```diff
@@ -547,7 +558,9 @@ bool GTP::execute(GameState & game, std::string xinput) {
             auto stonestring = game.board.get_stone_list();
             gtp_printf(id, "%s", stonestring.c_str());
         } else {
-            gtp_fail_printf(id, "Not a valid number of handicap stones");
+            gtp_fail_printf(id, 
+                "%d is not a valid number of handicap stones.\n"
+                "Use a number that is in the range 2 - 9", stones);
```
Comment (gcp) [inline src/GTP.cpp]: Free handicap allows more than 9 stones. Also if you get into this condition then the option was missing, so "stones" won't have a valid value.

File: src/GTP.cpp [hunk 4]
```diff
@@ -465,7 +470,9 @@ bool GTP::execute(GameState & game, std::string xinput) {
             } else if (color == "b" || color == "black") {
                 icolor = FastBoard::BLACK;
             } else {
-                gtp_fail_printf(id, "Color in time adjust not understood.\n");
+                gtp_fail_printf(id, 
+                    "Color $s in time adjust not understood.\n"
```
Comment (gcp) [inline src/GTP.cpp]: The command is time_left, not time adjust.

**Comment by gcp** [comment]
The user isn't supposed to type GTP commands. I'm not sure there is much point to this.

**Comment by gcp** [comment]
I'll close this because it's just very unlikely I'll end up merging it. GTP is meant to be used by developers that are working on some kind of interface. I am not sure these error messages help much there.

**Comment by barrybecker4** [comment]
OK. I didn't realize that the spec specified the actual error message content and the the GTP interface was supposed to take the GTP error message and translate it into something else that was more user friendly.

**Pull request closed without merging**